### PR TITLE
Improvement/add caching layer

### PR DESCRIPTION
**Pull request by tadovas**
Feel free to comment.
Currently only glide itself is cached ( we gained 0.5 sec :D )

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,24 @@ language: go
 go:
   - 1.9.2
 go_import_path: github.com/mysterium/node
+
+env:
+  global:
+    - CACHE_ROOT_DIR=$HOME/cache
+    - TRAVIS_TOOLS_PATH=$CACHE_ROOT_DIR/travis_tools
```
Comment (donce) [inline .travis.yml]: This is confusing - `TRAVIS_TOOLS_PATH` refers like it's tools given by Travis.
`BUILD_TOOLS_PATH` ?

File: bin/travis_scripts/ensure_glide [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+glide --version
+if [ $? -eq 0 ]; then
+    echo "Glide found"
+else
+    export GLIDE_VERSION="v0.13.1"
+    wget "https://github.com/Masterminds/glide/releases/download/$GLIDE_VERSION/glide-$GLIDE_VERSION-linux-amd64.tar.gz"
+    tar -vxz -C $TRAVIS_TOOLS_PATH --strip=1 -f glide-$GLIDE_VERSION-linux-amd64.tar.gz
+fi
```
Comment (donce) [inline bin/travis_scripts/ensure_glide]: newline missing 😵

File: .travis.yml (hunk 1 above)
Comment (tadovas) [reply .travis.yml]: fixed

File: bin/travis_scripts/ensure_glide (hunk 2 above)
Comment (tadovas) [reply bin/travis_scripts/ensure_glide]: added

**Pull request merged**